be/src/exec/vectorized/aggregate/agg_hash_map.h

```diff
@@ -119,64 +119,90 @@ struct AggHashMapWithOneNumberKey : public AggHashMapWithKey<HashMap> {
     AggDataPtr get_null_key_data() { return nullptr; }
 
     // prefetch branch better performance in case with larger hash tables
-    template <typename Func>
-    void compute_agg_prefetch(ColumnType* column, Buffer<AggDataPtr>* agg_states, Func&& allocate_func) {
+    template <typename Func, bool allocate_and_compute_state, bool compute_not_founds>
+    void compute_agg_prefetch(ColumnType* column, Buffer<AggDataPtr>* agg_states, Func&& allocate_func,
```

Comment:
This function is single-purpose in the past, why now need to be multiple-purpose? 
if we add another two single-purpose functions to support 1.allocate_and_compute_state && compute_not_founds  and 2. compute_not_founds, is it be better?

Reply:
> This function is single-purpose in the past, why now need to be multiple-purpose?
- there are many duplicated codes for computing allocate_and_compute_state and compute_not_founds seperately;
- add this template parameter doesn't add any affects with old implements;

> if we add another two single-purpose functions to support 1.allocate_and_compute_state && compute_not_founds and 2. compute_not_founds, is it be better?
- as above, we can use one function to express the purpose. use one function is better?